### ISSUE TITLE
flake8 flag F811 when ruff not flag it

### ISSUE THREAD

**Issue by lrandrianasoloarinavalona** [astral-sh/ruff#13324]
Hi, I don't know if it's one of the deviated behavior from flake8 or a bug.

It seem that RUFF flag F811 when it's a duplicate import, but weirdly, it don't work well with function. Here is a sample file.

```python
"""test."""


def _get_rejection_causes_for_image(image: str) -> set:
    causes = set()
    del image
    return causes


def _get_rejection_causes_for_image(image: str) -> set:
    causes = set()
    del image
    return causes


_get_rejection_causes_for_image("foo")

```

**Current behavior :**

Well `ruff check .` pass, flake8 flag F811

**Excpected Behavior :**

Ruff mark this as F811

**Ruff version :**

```sh
root@e0723ae7ba6c:/repo# ruff --version
ruff 0.6.4
```

**Comment by dhruvmanila** [comment]
Yeah, this is because the rule would ignore names like `_foo` (prefixed with one or more underscores) and others as per the [`dummy-variable-rgx`](https://docs.astral.sh/ruff/settings/#lint_dummy-variable-rgx) config value. I'll close this as it's working as expected, but feel free to ask any follow up questions that you might have.

**Comment by lrandrianasoloarinavalona** [comment]
Okay, thank you. I've used to use underscore to mark a function/method as private.
